codalab/lib/bundle_cli.py

```diff
@@ -1170,26 +1185,134 @@ def derive_bundle(self, bundle_type, command, targets, metadata):
+
+            uuid = generate_uuid()
+            bundle_path = '/opt/{}'.format(uuid)
+            command = args.command
+            request_network = None
+            dependencies = [
+                (u'{}'.format(key), u'/{}_dependencies/{}'.format(uuid, key)) for key, target in self.parse_target_specs(args.target_spec)
+            ]
+
+            # Set up a directory to store the bundle.
+            remove_path(bundle_path)
+            os.mkdir(bundle_path)
+
+            for dependency_path, docker_dependency_path in dependencies:
+                child_path = os.path.join(bundle_path, dependency_path)
+                os.symlink(docker_dependency_path, child_path)
+
+            dc = DockerClient()
+            container_id = dc.start_container(bundle_path, uuid, command, docker_image, request_network, dependencies)
+            print >>self.stdout, '===='
+            print >>self.stdout, 'ContainerID: ', container_id
+            print >>self.stdout, 'Local Bundle ID: ', uuid
+            print >>self.stdout, 'You can find local bundle contents in: ', bundle_path
+            print >>self.stdout, '===='
+        else:
+            targets = self.parse_key_targets(client, worksheet_uuid, args.target_spec)
+            new_bundle = client.create(
+                'bundles',
+                self.derive_bundle(RunBundle.BUNDLE_TYPE, args.command, targets, metadata),
+                params={'worksheet': worksheet_uuid},
+            )
+
+            print >>self.stdout, new_bundle['uuid']
+            self.wait(client, args, new_bundle['uuid'])
+
+    @Commands.command(
+        'edit-image',
+        help='Drop to an interactive container within an image as a bash shell to allow edits.',
+        arguments=(
+            Commands.Argument('target_spec', help=ALIASED_TARGET_SPEC_FORMAT, nargs='*', completer=TargetsCompleter),
+            Commands.Argument('--request-docker-image', help='The docker image to edit', required=True),
         )
+    )
+    def do_edit_image_command(self, args):
+        docker_image = args.request_docker_image
+
+        uuid = generate_uuid()
+        bundle_path = '/opt/{}'.format(uuid)
+        command = '/bin/bash'
+        request_network = None
+        dependencies = [
+            (u'{}'.format(key), u'/{}_dependencies/{}'.format(uuid, key)) for key, target in self.parse_target_specs(args.target_spec)
+        ]
+
+        # Set up a directory to store the bundle.
+        remove_path(bundle_path)
+        os.mkdir(bundle_path)
+
+        for dependency_path, docker_dependency_path in dependencies:
+            child_path = os.path.join(bundle_path, dependency_path)
+            os.symlink(docker_dependency_path, child_path)
+
+        dc = DockerClient()
+        extra_args = ['-it']
+        volume_bindings = dc.get_volume_bindings(bundle_path, uuid, command, docker_image, request_network, dependencies)
```

Comment:
All this logic I think should go in DockerClient.  Otherwise, we're exposing too much of the docker internals.